### PR TITLE
[13.0] [ADD] stock_lot_filter_available: New module to filter lots on stock

### PR DESCRIPTION
**Pull request by CarlosRoca13**
cc @Tecnativa TT29598

please @sergio-teruel @joao-p-marques review this

### CODE REVIEW

File: stock_lot_filter_available/i18n/es.po [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+#. module: stock_lot_filter_available
+#: model:ir.model,name:stock_lot_filter_available.model_stock_production_lot
+msgid "Lot/Serial"
+msgstr "Lote/Número de serie"
+
+#. module: stock_lot_filter_available
+#: model_terms:ir.ui.view,arch_db:stock_lot_filter_available.search_product_lot_filter
+msgid "On hand"
+msgstr "En stock"
```
Comment (sergio-teruel) [inline stock_lot_filter_available/i18n/es.po]: Odoo tranlates this term by "A mano"

File: stock_lot_filter_available/i18n/es.po [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+msgstr "Cantidad"
+
+#. module: stock_lot_filter_available
+#: code:addons/stock_lot_filter_available/models/stock_production_lot.py:0
+#, python-format
+msgid "Unsupported search operator"
+msgstr "Operador de búsqueda no soportado"
+
+#~ msgid "On stock"
+#~ msgstr "Disponible en inventario"
```
Comment (sergio-teruel) [inline stock_lot_filter_available/i18n/es.po]: Remove this

File: stock_lot_filter_available/models/stock_production_lot.py [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+
+    @api.model
+    def _is_unsupported_search_operator(self, operator):
+        return operator != ">"
+
+    def _search_product_qty(self, operator, value):
+        if self._is_unsupported_search_operator(operator):
+            raise ValueError(_("Unsupported search operator"))
+        ids = []
+        for lot in self.search([]):
```
Comment (sergio-teruel) [inline stock_lot_filter_available/models/stock_production_lot.py]: Maybe is better do that with a sql or readr_group to perform

**Review by sergio-teruel — CHANGES_REQUESTED** [review]

**Comment by CarlosRoca13** [comment]
@sergio-teruel changes done

File: stock_lot_filter_available/models/stock_production_lot.py [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+                and product_qty != value
+                or operator == ">"
+                and product_qty > value
+                or operator == ">="
+                and product_qty >= value
+                or operator == "<"
+                and product_qty < value
+                or operator == "<="
+                and product_qty <= value
+            ):
```
Comment (joao-p-marques) [inline stock_lot_filter_available/models/stock_production_lot.py]: Could you use `eval()` or similar to simplify this (make it prettier and more maintainable :smiley:)?
https://docs.python.org/3/library/functions.html#eval
https://github.com/odoo/odoo/blob/13.0/odoo/tools/safe_eval.py#L307

Comment (pedrobaeza) [reply stock_lot_filter_available/models/stock_production_lot.py]: The usual way of doing this is simply to pass the operator and value in another ORM search for getting the results.

**Comment by CarlosRoca13** [comment]
Done, but I limmited the search to the use case, for avoid using `safe_eval`

**Review by sergio-teruel — APPROVED** [review]
Functional testing ok

**Review by pedrobaeza — APPROVED** [review]

**Pull request merged**